Add `server info/list` (/s)

kristofbolyai opened the pull request:
The ported version of this feature functions slightly differently than the 1.12.2 implementation, but does everything without unnecessary options (like group and sort).

Screenshots of the feature:

`/s i` (or `/server i` or `/server info`)
![image](https://user-images.githubusercontent.com/49001742/156021719-9d53e7bf-6a04-497c-933a-ac0b4e152495.png)

`/s i 1` (or `/server i WC1` or `/server info 1`)
![image](https://user-images.githubusercontent.com/49001742/156021767-3eedfd48-05f9-4188-b7ac-817460c54cce.png)

`/s l` (or `/server list`)
![image](https://user-images.githubusercontent.com/49001742/156021860-93b0bea9-a358-4567-be59-1d3eca84ac46.png)

kristofbolyai committed:
Add `server info/list` (/s)

Incompleteusern commented on common/src/main/java/com/wynntils/commands/ServerCommand.java:
```diff
@@ -0,0 +1,196 @@
+                Commands.literal("server")
+                        .then(Commands.literal("list").executes(this::serverList))
+                        .then(Commands.literal("ls").executes(this::serverList))
+                        .then(Commands.literal("l").executes(this::serverList))
```
redirects maybe?

Incompleteusern commented on common/src/main/java/com/wynntils/commands/ServerCommand.java:
```diff
@@ -0,0 +1,196 @@
+                                                Commands.argument(
+                                                                "server", StringArgumentType.word())
+                                                        .executes(this::serverInfo))
+                                        .executes(this::serverInfoHelp))
```
same comment for redirects for this

Incompleteusern commented on common/src/main/java/com/wynntils/commands/ServerCommand.java:
```diff
@@ -0,0 +1,196 @@
+                        .executes(this::serverHelp));
+
+        dispatcher.register(
+                Commands.literal("s")
```
This definitely should be a redirect

Incompleteusern commented on common/src/main/java/com/wynntils/core/webapi/WebManager.java:
```diff
@@ -181,6 +187,41 @@ private static boolean tryReloadApiUrls(boolean async) {
+                "Mozilla/5.0 (Macintosh; U; Intel Mac OS X 10.4; en-US; rv:1.9.2.2) Gecko/20100316"
+                        + " Firefox/3.6.2");
+        // st.setRequestProperty("apikey", apiUrls.get("WynnApiKey")); API key rate limits
+        // constantly, perhaps we don't need it?
```
What does this comment mean?

kristofbolyai replied:
Api key rate limits constantly because it is shared between all wynntils users, but if we dont use  it then our ip has a separate rate limit

Incompleteusern replied:
iirc there's a comment in the discord which explains the usage of the api key for wynntils clients. I think its client specific too

kristofbolyai replied:
Not sure, with the Api key it always seemed rate limited, with code 429 (too many requests)

Incompleteusern commented on common/src/main/java/com/wynntils/core/webapi/WebManager.java:
```diff
@@ -181,6 +187,41 @@ private static boolean tryReloadApiUrls(boolean async) {
+        // st.setRequestProperty("apikey", apiUrls.get("WynnApiKey")); API key rate limits
+        // constantly, perhaps we don't need it?
+        st.setConnectTimeout(REQUEST_TIMEOUT_MILLIS);
+        st.setReadTimeout(REQUEST_TIMEOUT_MILLIS);
```
This is a later thing but connection generation should be separated

Incompleteusern reviewed (changes requested):
Some things

Incompleteusern committed:
add redirects

magicus commented on common/src/main/java/com/wynntils/commands/ServerCommand.java:
```diff
@@ -0,0 +1,180 @@
+
+public class ServerCommand extends WynntilsCommandBase {
+    private final List<String> serverTypes =
+            Lists.newArrayList("WC", "lobby", "GM", "DEV", "WAR", "HB");
```
I think this kind of data should be moved to the WC model. I have no specific suggestion, it's just a general note. We want to avoid ending up with the duplication of handling the WC model that plagues the current Wynntils code base. The way to achieve this is to never let actual "features" (which the commands actually are) handle specific knowledge about Wynncraft.

kristofbolyai committed:
Move serverTypes to WynnUtils

kristofbolyai committed:
Merge remote-tracking branch 'origin/command_server' into command_server

Incompleteusern committed:
move api generation code

Incompleteusern committed:
Merge branch 'command_server' of https://github.com/kristofbolyai/Artemis into pr/24

Incompleteusern committed:
Merge branch 'main' into pr/24

Incompleteusern committed:
fix

Incompleteusern committed:
spotless

Incompleteusern committed:
rewrite reloading for a feature based view instead

Also stop multiple attempts by WebManager to create a Wynntils Account, let user do so if needed

Incompleteusern committed:
Add Client Command Init

Incompleteusern committed:
spotless

magicus commented on common/src/main/java/com/wynntils/core/features/Feature.java:
```diff
@@ -18,9 +18,14 @@
 
     protected boolean enabled = false;
 
+    protected Feature(String name) {
```
I don't really like this. :( Having an argument-less constructor for the Feature made it so much nicer. 

I think it would be better to have an abstract getName() method that the Feature subclasses implement. There seems to be no compelling reason to put this literal string in the constructor,

Incompleteusern replied:
Ill do that then

Incompleteusern committed:
use abstract method instead

Incompleteusern committed:
fix issue

magicus reviewed (approved):
Much better, thanks!!!
